File: Cpp/11.cpp
//双指针问题
class Solution {
public:
    int maxArea(vector<int>& height) {
        int ans = 0, area;
        int l = 0, r = height.size() - 1;
        while(l < r) {
            area = min(height[l], height[r]) * (r - l);
            ans = max(ans, area);
            if(height[l] < height[r]) {
                l++;
            } else {
                r--;
            }
        }
        return ans;

    }
};

File: JavaScript/11.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    let l = 0, r = height.length - 1;
    let ans = 0;
    while(l < r) {
        let tmp = Math.min(height[l],height[r]) * (r-l);
        ans = Math.max(tmp, ans);
        if(height[l] < height[r]) {
            l++;
        } else {
            r--;
        }
    }
    return ans;
};
//对撞指针